projects/igniteui-angular/src/lib/date-common/util/date-time.util.ts

```diff
@@ -506,6 +529,58 @@ export abstract class DateTimeUtil {
         return false;
     }
 
+    public static isFormatNumeric(locale: string, inputFormat: string): boolean {
+        const dateParts = DateTimeUtil.parseDateTimeFormat(inputFormat);
+        if (predefinedNonNumericFormats.has(inputFormat) || dateParts.every(p => p.type === DatePart.Literal)) {
+            return false;
+        }
+        for (let i = 0; i < dateParts.length; i++) {
+            if (dateParts[i].type === DatePart.AmPm || dateParts[i].type === DatePart.Literal) {
+                continue;
+            }
+            const transformedValue = formatDate(new Date(), dateParts[i].format, locale);
+            // check if the transformed date/time part contains any kind of letter from any language
+            if (/\p{L}+/gu.test(transformedValue)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns an input format that can be used by the date-time editors, as
+     * - if the input format is already numeric, return it as is
+     * - if it is among the predefined numeric ones, return it as the equivalent locale-based format
+     *   for the corresponding numeric date parts
+     * - otherwise, return an empty string
+     */
+    public static getNumericInputFormat(locale: string, inputFormat: string): string {
```

Comment:
Would actually rename the `inputFormat` param to avoid confusing it with the return :) Perhaps just `format`?
And the doc comment might want some rewording following the same logic.